xocto/ranges.py

```diff
@@ -473,8 +484,13 @@ class HalfFiniteRange(Range[T]):
     HalfFiniteRange.
     """
 
+    __slots__ = ()
+
+    # Redefine types in base class
     start: T
-    boundaries = RangeBoundaries.INCLUSIVE_EXCLUSIVE
```

Comment:
Setting the default value using a class attribute does not play with `__slots__`. I have moved this default to `__init__`.